### PR TITLE
feat: find timetable pattern

### LINKED ISSUE #760: Add the Find a timetable pattern which is used on the new homepage
## Documentation

# Find a timetable
## What does it do?
* Helps users to find a timetable for their chosen service

## When to use it?
* On the homepage

## How it works?
* This will direct users to the timetables page
* Users can only select one travel mode
* Users will need to enter a service number if they select bus. If train or tram are selected, then users can go straight to the timetables page through the [Call to action button](https://designsystem.wmnetwork.co.uk/components/buttons/).

### PR DESCRIPTION
**Pull request by catiarodriguescosta**
Fixes #760

### CODE REVIEW

**Review by mrmjprice — CHANGES_REQUESTED** [review]
Please can you find a replace ```Find timetable``` and replace it with ```Find a timetable``` to add an 'a'? Thank you.

File: src/www/data.njk.json [hunk 1]
```diff
@@ -227,6 +227,9 @@
         {
           "name": "Feedback loop"
         },
+        {
+          "name": "Find timetable"
+        },
```
Comment (gldgrnt) [inline src/www/data.njk.json:232]:
```suggestion
        {
          "name": "Find a timetable"
        },
```

File: src/wmnds/patterns/find-a-timetable-widget/_find-a-timetable-widget.njk [hunk 2]
```diff
@@ -0,0 +1,108 @@
+{% macro wmndsFindATimetableWidget(params) %}
+
+{# Imports of components #}
+{% from "wmnds/components/content-card/_content-card.njk" import wmndsContentCard %}
+{% from "wmnds/components/form-elements/text-input/_text-input.njk" import wmndsTextInput %}
+{% from "wmnds/components/button/_button.njk" import wmndsButton %}
+{% from "wmnds/components/link/as-button/_as-button.njk" import wmndsLinkAsButton %}
+
+{# Set params #}
+{% set formTitle = params.formTitle if params.formTitle else "Find timetable" %}
```
Comment (gldgrnt) [inline src/wmnds/patterns/find-a-timetable-widget/_find-a-timetable-widget.njk:10]:
```suggestion
{% set formTitle = params.formTitle if params.formTitle else "Find a timetable" %}
```

File: .vscode/settings.json [hunk 3]
```diff
@@ -17,5 +17,22 @@
...
+    "activityBarBadge.foreground": "#15202b",
+    "statusBar.background": "#1857a4",
+    "statusBar.foreground": "#e7e7e7",
+    "statusBarItem.hoverBackground": "#1f6fd0",
+    "titleBar.activeBackground": "#1857a4",
+    "titleBar.activeForeground": "#e7e7e7",
+    "titleBar.inactiveBackground": "#1857a499",
+    "titleBar.inactiveForeground": "#e7e7e799"
+  },
+  "peacock.color": "#1857a4"
```
Comment (gldgrnt) [inline .vscode/settings.json:37]: Not sure if this should be committed (?)

**Review by gldgrnt — CHANGES_REQUESTED** [review]
All good with markup/styles, but title and page link are 'Find ticket' instead of 'Find a ticket' (and this causes a 404 when you click the link in the navigation or sidebar)

**Review by houbly — APPROVED** [review]
Looks good

**Review by jkmnch — CHANGES_REQUESTED** [review]
The train and tram buttons should be switched, so the order is Bus, Train and Tram.

Also, the 'Anything else' section is just for reference in the GitHub issue and shouldn't be in the DS.

**Review by jkmnch — CHANGES_REQUESTED** [review]
Sorry, last change! The Train button links to the correct timetables page on the old site, but the Tram button doesn't. Can the Tram button be linked to here? https://www.networkwestmidlands.com/plan-your-journey/timetables/#/?mode=Tram&serviceNo=MM1&limit=6

**Comment by catiarodriguescosta** [comment]
> Sorry, last change! The Train button links to the correct timetables page on the old site, but the Tram button doesn't. Can the Tram button be linked to here? https://www.networkwestmidlands.com/plan-your-journey/timetables/#/?mode=Tram&serviceNo=MM1&limit=6

No problem Jake, it's fixe now. Thanks for all the checks! :)

**Review by jkmnch — CHANGES_REQUESTED** [review]
The link doesn't work for me strangely. It reverts back to the default Timetables page when I click on the button. I don't think that link will work. It might have to be changed to this one instead: https://www.networkwestmidlands.com/plan-your-journey/timetables/#/route/cen_01MM1_%20_H_y11_26-26

**Review by jkmnch — DISMISSED** [review]
Looks great. Thanks Catia :)

**Review by gldgrnt — APPROVED** [review]
All looks good to me

**Review by houbly — APPROVED** [review]

**Review by mrmjprice — APPROVED** [review]

**Pull request merged**